### ISSUE TITLE
Additional feature: Image name as text variable for display

### ISSUE THREAD

**Issue by cjba850** [bitfocus/companion-module-osee-gostream#16]
Thanks for producing this.

A feature that would be nice quality of life thing (possibly needs manufacturer support)

ImageLibrary File Names visible on Still button test for simpler selection

When a user wants to switch to a different still - having the text (filename) as uploaded, can be a handy way to make it known what still you are switching to. This can be stored in the Gostream as the filename is provided on upload, and then the filename (minus extention) can be presented to companion as a variable, which can be used in the text display of the button. That's far more user friendly than Still 1/ Still2 / Still 3

workflow:
upload roadsign.jpg to ImageLibrary1
upload testPattern.jpg to ImageLibrary2
upload Cat.jpg to ImageLibrary3
upload dog.jpg to ImageLibrary4

Still 1 active slot has ImageLibrary 1 image attached
Still 2 active slot has ImageLibrary 3 image attached

Thus, $(gostreamdeck:ImageLibrary_1_Text) would be "roadsign"
Thus, $(gostreamdeck:ImageLibrary_2_Text) would be "testPattern"
Thus, $(gostreamdeck:ImageLibrary_3_Text) would be "Cat"
Thus, $(gostreamdeck:ImageLibrary_4_Text) would be "dog"

Thus, $(gostreamdeck:StillImage_Slot1_Text) would be "roadsign"
Thus, $(gostreamdeck:StillImage_Slot2_Text) would be "Cat"

and you can select the file (by name) for each slot, by assigning buttons and including the filename on the button text using those variables.

I looked, but couldn't see this in the current published release. Could have missed it though!

**Comment by cjba850** [comment]
Corollary: The Labels applied to each of the inputs (via the software) could also read from the names assigned to sources (there appears to be a way to label input 1 as camera1, input 2 as wide, input 3 as PC, etc, for the multiview display.)

That could also be handy on the input labels, already implemented in $(gostreamdeck:pre1_input_id), etc

Again, this may need manufacturer support as well.

**Comment by fa1k3n** [comment]
good idea. I have done a similar thing for selected video file in the v1.1.0 version. This can come in v1.2

**Comment by fa1k3n** [comment]
unfortunately the image names are not saved in the gostream device itself. So it is not possible to add them to variables.
